util/retry/retry.go

```diff
@@ -26,6 +27,82 @@ import (
 	"github.com/cockroachdb/cockroach/util/log"
 )
 
+type Config struct {
+	InitialInterval     time.Duration
+	MaxInterval         time.Duration
+	RandomizationFactor float64
+	Multiplier          float64
+	MaxRetries          int
+	Stopper             *util.Stopper
+}
+
+type Retry struct {
+	config         Config
+	currentAttempt int
+	isReset        bool
+}
+
+func Start(cfg Config) Retry {
+	if cfg.InitialInterval == 0 {
+		cfg.InitialInterval = 50 * time.Millisecond
+	}
+	if cfg.MaxInterval == 0 {
+		cfg.MaxInterval = 2 * time.Second
+	}
+	if cfg.RandomizationFactor == 0 {
+		cfg.RandomizationFactor = 0.15
+	}
+	if cfg.Multiplier == 0 {
+		cfg.Multiplier = 2
+	}
+
+	r := Retry{config: cfg}
+	r.Reset()
+	return r
+}
+
+func (r *Retry) Reset() {
+	r.currentAttempt = 0
+	r.isReset = true
+}
+
+func (r Retry) currentInterval() time.Duration {
```

Comment:
unless I'm wrong in my comment about renaming `InitialInterval`, this should be renamed to `currentBackoff()` and the local `interval` within accordingly, and the same for `MaxInterval`.